File: Source/StealFocus.Tracer.Web.UI/Test/MyChat.cs
namespace StealFocus.Tracer.Web.UI.Test
{
    using System;

    using SignalR.Hubs;

    [CLSCompliant(false)]
    [HubName("myChat")]
    public class MyChat : Hub
    {
        public void Send(string message)
        {
            // Call the addMessage method on all clients
            Clients.addMessage(message);
        }
    }
}

File: Source/StealFocus.Tracer.Web.UI/Test/MyChat.js
﻿/* File Created: February 20, 2012 */
var chat;

$(function () {
    // Created proxy
    chat = $.connection.myChat;

    // Assign a function to be called by the server
    chat.addMessage = onAddMessage;

    // Register a function with the button click
    $("#broadcast").click(onBroadcast);

    // Start the connection
    $.connection.hub.start();
});

function onAddMessage(message) {
    // Add the message to the list
    $('#messages').append('<li>' + message + '</li>');
}

function onBroadcast() {
    // Call the chat method on the server
    chat.send($('#message').val());
}